### PR TITLE
Bump to golang 1.18

### PR DESCRIPTION
**Pull request by bobbypage**
Fixes https://github.com/google/cadvisor/issues/3092

### CODE REVIEW

File: deploy/canary/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.17
+FROM golang:1.18
 MAINTAINER dashpole@google.com
```
Comment (SergeyKanzhelev) [inline deploy/canary/Dockerfile:2]: perhaps update it to your email

Comment (bobbypage) [reply deploy/canary/Dockerfile:2]: ack, will do in future PR!

**Comment by akhilerm** [comment]
@bobbypage Dont we need to bump the go.mod in the root and `cmd/` to 1.18 from 1.16?

**Comment by akhilerm** [comment]
I have raised a PR #3131 to update the module to 1.18

**Pull request merged**